misc/python/materialize/mzbuild.py

```diff
@@ -250,18 +317,36 @@ def build(self) -> None:
         cargo_profile = "release" if self.rd.release_mode else "debug"
 
         def copy(exe: Path) -> None:
-            (self.path / exe).parent.mkdir(parents=True, exist_ok=True)
-            shutil.copy(
-                self.rd.cargo_target_dir() / cargo_profile / exe, self.path / exe
-            )
+            exe_path = self.path / exe
+            dbg_path = exe_path.with_suffix(exe_path.suffix + ".debug")
+            exe_path.parent.mkdir(parents=True, exist_ok=True)
+            shutil.copy(self.rd.cargo_target_dir() / cargo_profile / exe, exe_path)
+
+            if self.split_debuginfo:
+                spawn.runv(
+                    [
+                        *self.rd.tool("objcopy"),
+                        exe_path,
+                        dbg_path,
+                        "--only-keep-debug",
+                    ],
+                    cwd=self.rd.root,
+                )
+                spawn.runv(
+                    [
+                        *self.rd.tool("objcopy"),
+                        exe_path,
+                        f"--add-gnu-debuglink={dbg_path}",
+                    ]
+                )
 
             if self.strip:
                 # NOTE(benesch): the debug information is large enough that it slows
                 # down CI, since we're packaging these binaries up into Docker
                 # images and shipping them around. A bit unfortunate, since it'd be
                 # nice to have useful backtraces if the binary crashes.
```

Comment:
Comment is stale. Nothing currently uses `strip: true` anymore. And the new logic doesn't have the same problem, because we split out the debug info!

```suggestion
```